src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -182,11 +180,34 @@ impl SharedBufferBatch {
     }
 
     pub fn start_user_key(&self) -> &[u8] {
-        key::user_key(&self.inner.first().unwrap().0)
+        if !self.inner.delete_range_tombstones.is_empty()
+            && (self.inner.is_empty()
+                || self
+                    .inner
+                    .delete_range_tombstones
+                    .first()
+                    .unwrap()
+                    .start_user_key
+                    .as_slice()
+                    .le(key::user_key(&self.inner.first().unwrap().0)))
+        {
+            self.inner
+                .delete_range_tombstones
+                .first()
+                .unwrap()
+                .start_user_key
+                .as_slice()
+        } else {
+            key::user_key(&self.inner.first().unwrap().0)
+        }
     }
 
     pub fn end_user_key(&self) -> &[u8] {
```

Comment:
I think the semantic of method is unclear because it sometimes returns an inclusive end_user_key and sometimes returns an exclusive end_user_key. Will this be a problem?

Reply:
Yes, you are write, let me see....